core/kover/learning/experiment.py

```diff
@@ -437,7 +440,27 @@ def normal_raise(exception):
     p = np.unique(p)
 
     dataset = KoverDataset(dataset_file)
-
+    
```

Comment:
This new chunk of code should be in a separate function. It has it's own purpose.